src/dattobd.c

```diff
@@ -4229,6 +4311,52 @@ static long ctrl_ioctl(struct file *filp, unsigned int cmd, unsigned long arg){
 		}
 
 		break;
+    case IOCTL_DATTOBD_VERSION:
+        src = DATTOBD_VERSION;
+        dest = ((struct dattobd_version __user *)arg)->version_string;
+        len = strlen(src) + 1;
+        vslen = sizeof((struct dattobd_version *)0)->version_string;
+        // make sure we don't copy more than the user should have allocated.
+        if (len > vslen)
+            len = vslen;
+        ret = copy_to_user(dest, src, len);
+        // if the string is too long, make sure to nt the last position of the user's buffer.
+        if ((ret == 0) && (len == vslen))
```

Comment:
it's gone now, but it does, if the first copy worked and the string fills the buffer, then write a zero at the last position.